src/middlewares/__tests__/request-authorization.test.js

```diff
@@ -0,0 +1,106 @@
+const requestAuthorization = require('../request-authorization');
+const Request = require('../../models/request');
+
+const trueOwnerId = '123456';
+const falseOwnerId = '654321';
+const correctRequestId = 'correctId';
+const incorrectRequestId = 'incorrectId';
+const request = {
+  _id: 'correctId',
+  owner: '123456',
+};
+const res = {
+  status: jest.fn().mockReturnThis(),
+  json: jest.fn().mockReturnThis(),
+};
+jest.mock('../../models/request', () => ({
+  findById: jest.fn().mockImplementation((id) => {
+    if (id === request._id) {
+      return Promise.resolve(request);
+    }
+    return Promise.resolve(null);
+  }),
+}));
+afterAll(() => {
+  jest.clearAllMocks();
+});
+describe('request-authorization function middleware ', () => {
+  test('Should authorize when requesting user is the owner of request', async () => {
+    const req = {
+      params: {
+        id: correctRequestId,
+      },
+      user: {
+        _id: trueOwnerId,
+      },
+    };
+
+    const next = jest.fn();
+    await requestAuthorization(req, res, next);
+    expect(Request.findById).toHaveBeenCalled();
+    expect(Request.findById).toHaveBeenCalledWith(correctRequestId);
+    expect(next).toHaveBeenCalled();
+    expect(res.status).not.toHaveBeenCalled();
+    expect(res.json).not.toHaveBeenCalled();
+  });
+
+  test('Should unauthorize when requesting user is not the owner of request', async () => {
+    const req = {
```

Comment:
I think a better test statement would be `Should not authorize when the requesting user is not the owner of the request`

Reply:
same here

Reply:
Don't mind these I don't have an excellent English level 😅 
I just felt it would be better, not necessarily true 😁

Reply:
I kept them because Shrreya went over that test and didn't say anything.I don't trust my English either  😁